### PR TITLE
Add CategoryForeignKey and CategoryOneToOneField fields.

### PR DESCRIPTION
**Pull request by chronossc**
Add CategoryForeignKey and CategoryOneToOneField fields.

CategoryOneToOneField should be used to extend Categories because a issue in treebeard [1] that blocks multi-table inheritance.

CategoryForeignKey can be used when you need only one category to an object instead many.

[1] http://code.tabo.pe/django-treebeard/issue/80/get_result_class-when-creating-new-object

### CODE REVIEW

**Comment by mkoistinen** [comment]
Anything you can do to address the fact that coverage has dropped a whopping 10% for this PR?

File: aldryn_categories/fields.py [hunk 1]
```diff
@@ -15,6 +15,56 @@
...
+    # This is necessary for Django 1.7.4+
+    def get_internal_type(self):
+        return 'ForeignKey'
+
+    def formfield(self, form_class=CategoryModelChoiceField,
+                  choices_form_class=None, **kwargs):
+        kwargs["form_class"] = form_class
+        kwargs["choices_form_class"] = choices_form_class
+        return super(OneToOneField, self).formfield(**kwargs)
+
```
Comment (mkoistinen) [inline aldryn_categories/fields.py]: What is the use-case for a CategoryOneToOneField?

**Comment by intellectronica** [comment]
And the tests are where?

**Comment by mkoistinen** [comment]
After discussion with @intellectronica, we agreed that this can be committed now, but we'll need to add tests later.

**Pull request merged**